### PR TITLE
Code enhancement and cleanup

### PR DESCRIPTION
**Pull request by yceruto**
| Q | A |
| --- | --- |
| Branch? | 2.7 |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/Console/Tests/Helper/HelperSetTest.php [hunk 1]
```diff
@@ -113,6 +113,8 @@ public function testIteration()
      *
      * @param string    $name
      * @param HelperSet $helperset allows a mock to verify a particular helperset set is being added to the Helper
+     *
+     * @return \PHPUnit_Framework_MockObject_MockObject
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Tests/Helper/HelperSetTest.php]: HelperInterface would be more useful

Comment (fabpot) [inline src/Symfony/Component/Console/Tests/Helper/HelperSetTest.php]: We do not document tests, so the whole phpdoc might be removed.

File: src/Symfony/Component/Debug/ErrorHandler.php [hunk 2]
```diff
@@ -349,14 +349,17 @@ private function reRegister($prev)
...
+     * @param string $message
      * @param string $file
      * @param int    $line
      * @param array  $context
+     * @param array  $backtrace
      *
      * @return bool Returns false when no handling happens so that the PHP engine can handle the error itself
      *
-     * @throws \ErrorException When $this->thrownErrors requests so
+     * @throws ContextErrorException
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Debug/ErrorHandler.php]: ContextErrorException inherits from \ErrorException, thus should be removed

File: src/Symfony/Component/HttpFoundation/JsonResponse.php [hunk 3]
```diff
@@ -91,7 +102,7 @@ public function setCallback($callback = null)
      *
      * @return JsonResponse
      *
-     * @throws \InvalidArgumentException
+     * @throws \Exception
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/JsonResponse.php]: re-throwing is not throwing, should be kept asis

File: src/Symfony/Component/VarDumper/Cloner/AbstractCloner.php [hunk 4]
```diff
@@ -164,6 +164,8 @@ public function setMaxString($maxString)
      * @param int   $filter A bit field of Caster::EXCLUDE_* constants
      *
      * @return Data The cloned variable represented by a Data object
+     *
+     * @throws \Exception
```
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Cloner/AbstractCloner.php]: re-throwing is not throwing, should be removed

File: src/Symfony/Component/VarDumper/Dumper/AbstractDumper.php [hunk 5]
```diff
@@ -131,6 +131,8 @@ public function setIndentPad($pad)
      *
      * @param Data                          $data   A Data object
      * @param callable|resource|string|null $output A line dumper callable, an opened stream or an output path
+     *
+     * @throws \Throwable
```
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Dumper/AbstractDumper.php]: same

File: src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php [hunk 6]
```diff
@@ -110,6 +110,7 @@ public function onKernelResponse(FilterResponseEvent $event)
      * Injects the web debug toolbar into the given Response.
      *
      * @param Response $response A Response instance
+     * @param Request  $request  A Request instance
```
Comment (fabpot) [inline src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php]: Instead, I would remove the previous @param as these phpdocs do not add any values.

File: src/Symfony/Component/Asset/PathPackage.php [hunk 7]
```diff
@@ -31,6 +31,7 @@ class PathPackage extends Package
     /**
      * @param string                   $basePath        The base path to be prepended to relative paths
      * @param VersionStrategyInterface $versionStrategy The version strategy
+     * @param ContextInterface|null    $context         Context
```
Comment (fabpot) [inline src/Symfony/Component/Asset/PathPackage.php]: Should be reverted as it does not add any value over the method signature

File: src/Symfony/Component/Console/Command/Command.php [hunk 8]
```diff
@@ -569,6 +569,8 @@ public function getSynopsis($short = false)
      * Add a command usage example.
      *
      * @param string $usage The usage, it'll be prefixed with the command name
+     *
+     * @return Command The current instance
```
Comment (fabpot) [inline src/Symfony/Component/Console/Command/Command.php]: You should check but I think we are using `@return $this` elsewhere

File: src/Symfony/Component/Console/Helper/Table.php [hunk 9]
```diff
@@ -282,6 +282,8 @@ private function renderRow(array $row, $cellFormat)
      * @param array  $row
      * @param int    $column
      * @param string $cellFormat
+     *
+     * @return string
```
Comment (fabpot) [inline src/Symfony/Component/Console/Helper/Table.php]: Not something we do except if this is really unexpected, but that's not the case here.

File: src/Symfony/Component/Console/Command/Command.php (hunk 8 above)
Comment (yceruto) [inline src/Symfony/Component/Console/Command/Command.php]: In this class `@return Command` is used, should change this to `return $this` in the whole class ?

**Comment by nicolas-grekas** [comment]
:+1:

File: src/Symfony/Component/Console/Command/Command.php (hunk 8 above)
Comment (HeahDude) [inline src/Symfony/Component/Console/Command/Command.php]: `return self` is valid as well as `$this` for phpdoc but the first one is more consistent with the future 7.1 type hinting feature.

**Comment by javiereguiluz** [comment]
👍

File: src/Symfony/Component/HttpFoundation/JsonResponse.php [hunk 10]
```diff
@@ -50,7 +50,18 @@ public function __construct($data = null, $status = 200, $headers = array())
     }
 
     /**
-     * {@inheritdoc}
+     * Factory method for chainability.
+     *
+     * Example:.
```
Comment (fabpot) [inline src/Symfony/Component/HttpFoundation/JsonResponse.php]: extra dot here

File: src/Symfony/Component/HttpFoundation/Tests/ResponseHeaderBagTest.php [hunk 11]
```diff
@@ -253,6 +253,7 @@ public function testToStringDoesntMessUpHeaders()
         $headers->set('Location', 'http://www.symfony.com');
         $headers->set('Content-type', 'text/html');
 
+        //invoking to __toString method
```
Comment (fabpot) [inline src/Symfony/Component/HttpFoundation/Tests/ResponseHeaderBagTest.php]: can be removed as this is just a description of the next line.

File: src/Symfony/Component/Templating/PhpEngine.php [hunk 12]
```diff
@@ -142,6 +142,7 @@ protected function evaluate(Storage $template, array $parameters = array())
             throw new \InvalidArgumentException('Invalid parameter (view)');
         }
 
+        // the view variables is exposed to the require file below
```
Comment (fabpot) [inline src/Symfony/Component/Templating/PhpEngine.php]: variable with an s

**Comment by yceruto** [comment]
Fixed @fabpot's comments :+1:

**Comment by nicolas-grekas** [comment]
Thank you @yceruto.

**Pull request closed without merging**